esmvalcore/preprocessor/_time.py

```diff
@@ -101,13 +111,24 @@ def extract_time(
     ------
     ValueError
         Time ranges are outside the cube time limits.
-
     """
+    if start_year is not None:
+        start_year = int(start_year)
+    if end_year is not None:
+        end_year = int(end_year)
+    if (not start_year) ^ (not end_year):
```

Comment:
```suggestion
    if (start_year is None) ^ (end_year is None):
```
to avoid errors when start_year or end_year == 0